addons/mail/static/src/js/text_emojis_field.js

```diff
@@ -0,0 +1,143 @@
+odoo.define('social.form_field_emoji', function (require) {
+"use strict";
+
+var basicFields = require('web.basic_fields');
+var core = require('web.core');
+var emojis = require('mail.emojis');
+var registry = require('web.field_registry');
+var MailEmojisMixin = require('mail.emoji_mixin');
+var _onEmojiClickMixin = MailEmojisMixin._onEmojiClick;
+var QWeb = core.qweb;
+
+/*
+ * Common code for FieldText and FieldChar
+ */
+var FieldEmojiCommon = {
```

Comment:
This is included in the MOVE commit, but it does a little more that juste move.

I think you should split in 2 commits:

- One that simply moves the FieldTextEmojis, and rename namespaces
- One that adds a FieldTextCommon and the 2 supported fields

This will make the history a lot cleaner when we try to understand what's going on.
Also, it will allow you to make more explicit commit messages.

Sorry, I know it's annoying, but it's important stuff.